### PR TITLE
Add Simple-WebSocket-Server 2.0.0

### PR DESCRIPTION
**Pull request by IceflowRE**
Specify library name and version:  **simple-websocket-server/2.0.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Note: I stayed with `simple-websocket-server` as package name, even if its misleading as it supports server and client.

### CODE REVIEW

File: recipes/simple-websocket-server/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+        
+    def requirements(self):
+        if self.options.use_asio_standalone:
+            self.requires("asio/1.13.0")
+        else:
+            raise ConanInvalidConfiguration("Is not supported yet. Required boost/1.54 is not available.")
```
Comment (uilianries) [inline recipes/simple-websocket-server/all/conanfile.py]: can you use a newer boost version? like 1.71 or 1.70

Comment (IceflowRE) [reply recipes/simple-websocket-server/all/conanfile.py]: Oh, my bad, i had another error while creating the package with the build step (removing the build from the package solved it) and i forgot about that option.

Comment (IceflowRE) [reply recipes/simple-websocket-server/all/conanfile.py]: I tested with boost/1.71.0 too.

File: recipes/simple-websocket-server/config.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+versions:
+  "v2.0.0":
```
Comment (uilianries) [inline recipes/simple-websocket-server/config.yml]:
```suggestion
  "2.0.0":
```

Please, discard the prefix

File: recipes/simple-websocket-server/all/conandata.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+sources:
+  "v2.0.0":
```
Comment (uilianries) [inline recipes/simple-websocket-server/all/conandata.yml]:
```suggestion
  "2.0.0":
```

Please, discard the prefix

**Comment by uilianries** [comment]
I can confirm that it is working with Boost.

File: recipes/simple-websocket-server/all/conandata.yml (hunk 3 above)
Comment (IceflowRE) [reply recipes/simple-websocket-server/all/conandata.yml]: Even if the main project used those?

File: recipes/simple-websocket-server/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            self.requires("asio/1.13.0")
+        else:
+            self.requires("boost/1.71.0")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "Simple-WebSocket-Server-v" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
```
Comment (danimtb) [inline recipes/simple-websocket-server/all/conanfile.py]: This is not used, maybe remove it? I will make the recipe more clear

File: recipes/simple-websocket-server/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+class SimpleWebSocketServerConan(ConanFile):
+    name = "simple-websocket-server"
+    homepage = "https://gitlab.com/eidheim/Simple-WebSocket-Server"
+    description = "A very simple, fast, multithreaded, platform independent WebSocket (WS) and WebSocket Secure (WSS) server and client library."
+    topics = ("websocket", "socket", "server", "client", "header-only")
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "compiler", "arch", "build_type"
+    no_copy_source = True
+    generators= "cmake"
```
Comment (uilianries) [inline recipes/simple-websocket-server/all/conanfile.py]:
```suggestion
```

File: recipes/simple-websocket-server/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
+from conans import ConanFile, tools, CMake
```
Comment (uilianries) [inline recipes/simple-websocket-server/all/conanfile.py]:
```suggestion
from conans import ConanFile, tools
```

**Pull request merged**